.gitignore

```diff
@@ -1,22 +1,102 @@
-# eclipse
-bin
-*.launch
-.settings
+#ant stuff
+/bin/
+/download/
+#Remove OS generated garbage
+*/.DS_Store
+.DS_Store
+.DS_Store?
+.Spotlight-V100
+.Trashes
+Icon?
+ehthumbs.db
+Thumbs.db
+#gradle stuff
+/.gradle
+/build/
+/run*/
+/ui/
+gradle-app.setting
+#IDEA files from Gradle
+.idea/
+/*.iml
+/*.ipr
+/*.iws
+#Vim backups
+*~
+#manual version override
+version.properties
+#eclipse stuffs
+/.classpath
+/.project
+/.settings/
+/eclipse/
+/debug/
+/.metadata/
+/config/
+/logs/
+options.txt
+/saves/
+/out
+/tmp
+.iws
+**/.idea/workspace.xml
+**/.idea/tasks.xml
+*.iml
+/classes/
 .metadata
-.classpath
-.project
+bin/
+tmp/
+*.tmp
+*.bak
+*.swp
+*~.nib
+local.properties
+.settings/
+.loadpath
+.recommenders
 
-# idea
-out
-*.ipr
-*.iws
-*.iml
-.idea
+# External tool builders
+.externalToolBuilders/
+
+# Locally stored "Eclipse launch configurations"
+*.launch
+
+# PyDev specific (Python IDE for Eclipse)
+*.pydevproject
```

Comment:
Why is there a python project ignore? Minecraft does not use python. Actually, a few of these .gitignore entries need a bit more explanation as to how they apply to this project, some should probably just be in your global .gitignore.